user/pages/03.commerce2/02.developer-guide/03.core/01.conditions/docs.md

```diff
@@ -4,8 +4,284 @@ taxonomy:
+
+  $form[***'operator'***] = [
+    '#type' => 'select',
+    '#title' => t('Operator'),
+    '#options' => $this->getComparisonOperators(),
+    '#default_value' => $this->configuration[***'operator'***],
+    '#required' => TRUE,
+  ];
+  
+  $form['quantity'] = [
+    '#type' => 'number',
+    '#title' => t('Quantity'),
+    '#default_value' => $this->configuration['quantity'],
+    '#min' => 1,
+    '#required' => TRUE,
+  ];
+
+  return $form;
+}
+```
+
+It is important not to hardcore values for a component's `'#default_value'` or 
+`'#value'` in the `buildConfigurationForm` function. Doing so will stop any new
+configuration values set by user input from being saved. Use 
+`defaultConfiguration` function to set default values. 
+
+
+Default Configuration Settings
+---
+
+Emphasis of setting name `***'operator'***` is to show how this name should be 
+used across all functions in this class. 
+
+```
+public function defaultConfiguration() {
+  return [
+    ***'operator'*** => '>',
+    'quantity' => 1,
+  ] + parent::defaultConfiguration();
+}
+```
+
+
+Submit Configuration Form
+---
+
+This is where user inputted values are saved to the form configuration settings.
+
+```
+public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+  parent::submitConfigurationForm($form, $form_state);
+
+  $values = $form_state->getValue($form['#parents']);
+  
+// Note that the setting name here needs to be consistent with defaultConfig
+// and buildConfigurationForm.
+  $this->configuration[***'operator'***] = $values[***'operator'***];
+  $this->configuration['quantity'] = $values['quantity'];
+}
+```
+
+
+Evaluate Function
+---
+
+This is the guts of a condition. The goal of `evaluate` is to decide if an 
+order meets a certain criteria and to return a boolean TRUE if so. 
+
+```
+// *** Remember annotations are inherited from ConditionInterface.***
+public function evaluate(EntityInterface $entity) {
+
+// This line is essential as it ensures this condition will only run for the 
+// 'entity_type' declared in this condition's annotations. 
+  $this->assertEntity($entity);
+  
+  /** @var \Drupal\commerce_order\Entity\OrderInterface $order */
+  $order = $entity;
+  /** @var \Drupal\commerce_promotion\Entity\PromotionInterface $promotion */
+  $promotion = $this->parentEntity;
+  $offer = $promotion->getOffer();
+
+  $quantity = '0';
+  foreach ($order->getItems() as $order_item) {
+  
+// First assess if each order item can apply to this condition.
+    // If the offer has conditions, skip order items that don't match.
+    if ($offer instanceof OrderItemPromotionOfferInterface) {
+    
+      $conditions = $offer->getConditions();
+      $condition_group = new ConditionGroup($conditions, 'OR');
+      
+      if (!$condition_group->evaluate($order_item)) {
+        continue;
+      }
+    }
+    
+// Get the quantity from each item and add to a running total.
```

Comment:
Not a fan of this indentation thing. If you want to have inline comments, break the code up into smaller blocks of code with your comment in between.